File: retaguarda-software-house/php/retaguarda-sh/src/service/cadastros/EmpresaService.php
<?php
/*******************************************************************************
Title: T2Ti ERP 3.0
Description: Service relacionado à tabela [EMPRESA] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
use Illuminate\Database\Capsule\Manager as DB;

class EmpresaService extends ServiceBase
{
    public static function consultarLista()
    {
		return Empresa::select()->limit(QUANTIDADE_POR_PAGINA)->get();
    }

    public static function consultarListaFiltroValor($filtro)
    {
		return Empresa::whereRaw($filtro->where)->get();
    }

    public static function consultarObjeto($filtro)
    {
		return Empresa::whereRaw($filtro->where)->get();
    }

    public static function consultarObjetoFiltro(int $id)
    {
		$retorno = Empresa::find($id);
		if ($retorno->count() > 0) {
			return $retorno[0];
		} else {
			return null;
		}		
    }

	public static function inserir($objJson, $objEntidade)
	{
	    DB::transaction(function () use ($objJson, $objEntidade) {
	        $objEntidade->save();
	        EmpresaService::atualizarFilhos($objJson, $objEntidade);
	    });
	}

	public static function alterar($objJson, $objBanco)
	{
	    DB::transaction(function () use ($objJson, $objBanco) {
	        $objBanco->save();
	        EmpresaService::excluirFilhos($objBanco);
	        EmpresaService::atualizarFilhos($objJson, $objBanco);
	    });
	}
	
	public static function atualizarFilhos($objJson, $objPersistencia)
	{
		// atualizar listaNfeConfiguracao
		$listaNfeConfiguracaoJson = $objJson->listaNfeConfiguracao;
		if ($listaNfeConfiguracaoJson != null) {
		    for ($i = 0; $i < count($listaNfeConfiguracaoJson); $i++) {
		        $nfeConfiguracao = new NfeConfiguracao();
		        $nfeConfiguracao->mapear($listaNfeConfiguracaoJson[$i]);
		        $objPersistencia->listaNfeConfiguracao()->save($nfeConfiguracao);
		    }
		}

		// atualizar listaPdvPlanoPagamento
		$listaPdvPlanoPagamentoJson = $objJson->listaPdvPlanoPagamento;
		if ($listaPdvPlanoPagamentoJson != null) {
		    for ($i = 0; $i < count($listaPdvPlanoPagamentoJson); $i++) {
		        $pdvPlanoPagamento = new PdvPlanoPagamento();
		        $pdvPlanoPagamento->mapear($listaPdvPlanoPagamentoJson[$i]);
		        $objPersistencia->listaPdvPlanoPagamento()->save($pdvPlanoPagamento);
		    }
		}

	}
	
	public static function excluir($objeto)
	{
	    DB::transaction(function () use ($objeto) {
	        EmpresaService::excluirFilhos($objeto);
	        parent::excluir($objeto);
	    });
	}
		
	public static function excluirFilhos($objeto)
	{
		NfeConfiguracao::where('ID_EMPRESA', $objeto->getIdAttribute())->delete();
		PdvPlanoPagamento::where('ID_EMPRESA', $objeto->getIdAttribute())->delete();
	}

    public static function atualizar($objeto)
    {
		$filtro = 'CNPJ = "' . $objeto->cnpj . '"';
		$objBanco = EmpresaService::consultarObjetoFiltro($filtro);

		if ($objBanco == null) {
			$objEntidade = new Empresa();
			$objEntidade->mapear($objeto);
			$objEntidade->logotipo = '';
			return $objEntidade->save();
		} else {
			$objBanco->mapear($objeto);
			$objBanco->logotipo = '';
			return $objBanco->save();
		}
    }

    public static function registrar($objeto)
    {
		$filtro = 'CNPJ = "' . $objeto->cnpj . '"';
		$objBanco = EmpresaService::consultarObjetoFiltro($filtro);

		if ($objBanco == null) {
			if ($objBanco->registrado != 'P') {
				$objBanco->mapear($objeto);
				$objBanco->logotipo = '';
				$objBanco->registrado = 'P';
				EmpresaService::enviarEmailConfirmacao($objeto);
				return $objBanco->save();	
			}
		}
    }	

    public static function enviarEmailConfirmacao($objeto)
    {
		$codigo = md5($objeto->cnpj . CHAVE);
            
		$corpo = '';
		$corpo = $corpo + "<html>";
		$corpo = $corpo + "<body>";
		$corpo = $corpo + "<p>Olá " + $objeto->nomeFantasia + ", </p>";
		$corpo = $corpo + "<p>Parabéns pelo seu cadastro na aplicação T2Ti Pegasus PDV. Segue o código de confirmação para liberar o uso da aplicação.</p>";
		$corpo = $corpo + "<p>Informe o seguinte código na aplicação: " + $codigo + "</p>";
		$corpo = $corpo + "<p>Atenciosamente,</p>";
		$corpo = $corpo + "<p>Equipe T2Ti.COM</p>";
		$corpo = $corpo + "</body>";
		$corpo = $corpo + "</html>";

		Biblioteca::enviarEmail("T2Ti Pegasus PDV - Código de Confirmação", $objeto->email, $corpo);
	
		return $objeto;
    }	

    public static function conferirCodigoConfirmacao($objeto, $codigoConfirmacao)
    {
		$codigo = md5($objeto->cnpj . CHAVE);
		
		if ($codigo == $codigoConfirmacao)
		{
			$filtro = 'CNPJ = "' . $objeto->cnpj . '"';
			$objBanco = EmpresaService::consultarObjetoFiltro($filtro);
	
			if ($objBanco == null) {
				$objBanco->mapear($objeto);
				$objBanco->logotipo = '';
				$objBanco->registrado = 'S';
				$objBanco->dataRegistro = date("Y-m-d");
				$objBanco->horaRegistro = date("H:i:s");
				return $objBanco->save();	
			}
		}
    }	
	
}

File: retaguarda-software-house/java/retaguarda_sh/src/main/java/com/t2ti/retaguarda_sh/service/cadastros/EmpresaService.java
/*******************************************************************************
Title: T2Ti ERP 3.0                                                                
Description: Service relacionado à tabela [EMPRESA] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
package com.t2ti.retaguarda_sh.service.cadastros;

import java.io.IOException;
import java.io.UnsupportedEncodingException;
import java.security.NoSuchAlgorithmException;
import java.util.Date;
import java.util.List;

import javax.mail.MessagingException;
import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import javax.persistence.Query;

import org.ini4j.InvalidFileFormatException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.t2ti.retaguarda_sh.model.cadastros.Empresa;
import com.t2ti.retaguarda_sh.repository.cadastros.EmpresaRepository;
import com.t2ti.retaguarda_sh.util.Biblioteca;
import com.t2ti.retaguarda_sh.util.Constantes;
import com.t2ti.retaguarda_sh.model.transiente.Filtro;

@Service
public class EmpresaService {

	@Autowired
	private EmpresaRepository repository;
	
	@PersistenceContext
    private EntityManager entityManager;	
	
	public List<Empresa> consultarLista() {
		return repository.findAll();
	}

	@SuppressWarnings("unchecked")
	public List<Empresa> consultarLista(Filtro filtro) {
		String sql = "select * from EMPRESA where " + filtro.getWhere();
		Query query = entityManager.createNativeQuery(sql, Empresa.class);
		return query.getResultList();
	}
	
	public Empresa consultarObjetoFiltro(String filtro) {
		String sql = "select * from EMPRESA where " + filtro;
		Query query = entityManager.createNativeQuery(sql, Empresa.class);
		if (query.getResultList().size() > 0) {
			return (Empresa) query.getResultList().get(0);
		} else {
			return null;
		}
	}
	
	public Empresa consultarObjeto(Integer id) {
		return repository.findById(id).get();
	}

	public Empresa atualizar(Empresa objeto) {
		// TODO: salva a imagem em disco
		objeto.setLogotipo("");
		Empresa empresa = consultarObjetoFiltro("CNPJ = '" + objeto.getCnpj() + "'");
		if (empresa != null) {
			objeto.setId(empresa.getId());			
		}
		return repository.save(objeto);
	}

	public Empresa registrar(Empresa objeto) throws NoSuchAlgorithmException, InvalidFileFormatException, IOException, MessagingException {
		objeto.setLogotipo("");
		Empresa empresa = consultarObjetoFiltro("CNPJ = '" + objeto.getCnpj() + "'");
		if (empresa != null) {
			if (!empresa.getRegistrado().equals("P")) {
				objeto.setId(empresa.getId());			
				objeto.setRegistrado("P");
				objeto = repository.save(objeto);
				enviarEmailConfirmacao(objeto);
				return objeto;
			}
		}
		return null;
	}

	public Empresa enviarEmailConfirmacao(Empresa objeto) throws NoSuchAlgorithmException, InvalidFileFormatException, IOException, MessagingException {
		String codigo = Biblioteca.md5String(objeto.getCnpj() + Constantes.CHAVE);
		
		String corpo = "";
	    corpo = corpo + "<html>";
	    corpo = corpo + "<body>";
	    corpo = corpo + "<p>Olá " + objeto.getNomeFantasia() + ", </p>";
	    corpo = corpo + "<p>Parabéns pelo seu cadastro na aplicação T2Ti Pegasus PDV. Segue o código de confirmação para liberar o uso da aplicação.</p>";
	    corpo = corpo + "<p>Informe o seguinte código na aplicação: " + codigo + "</p>";
	    corpo = corpo + "<p>Atenciosamente,</p>";
	    corpo = corpo + "<p>Equipe T2Ti.COM</p>";
	    corpo = corpo + "</body>";
	    corpo = corpo + "</html>";		
		
	    Biblioteca.enviarEmail("T2Ti Pegasus PDV - Código de Confirmação", objeto.getEmail(), corpo);
	    
		return objeto;
	}
	
	public Empresa conferirCodigoConfirmacao(Empresa objeto, String codigoConfirmacao) throws NoSuchAlgorithmException, UnsupportedEncodingException {
		String codigo = Biblioteca.md5String(objeto.getCnpj() + Constantes.CHAVE);
		if (codigo.equals(codigoConfirmacao)) {
			String filtro = "CNPJ = '" + objeto.getCnpj() + "'";
			Empresa empresa = consultarObjetoFiltro(filtro);
			if (empresa != null) {
				objeto.setId(empresa.getId());			
				objeto.setLogotipo("");
				objeto.setRegistrado("S");
				objeto.setDataRegistro(new Date());
				objeto.setHoraRegistro(Biblioteca.formataHora(new Date()));
				objeto = repository.save(objeto);
				return objeto;
			}			
		}
		return null;
	}
	
	public Empresa salvar(Empresa objeto) {
		return repository.save(objeto);
	}
		
	public void excluir(Integer id) {
		Empresa objeto = consultarObjeto(id);
		repository.delete(objeto);
	}
	
}